lib/bson/boolean.rb

```diff
@@ -37,7 +37,7 @@ class Boolean
     #
     # @since 2.0.0
     def self.from_bson(bson)
-      bson.readbyte == 1
+      bson.readbyte.chr.force_encoding('BINARY') == TrueClass::TRUE_BYTE
```

Comment:
What caused the need to `force_encoding` here? We should always be getting binary strings on reads.